code/datums/progressbar.dm

```diff
@@ -58,10 +60,8 @@
 	bars.Remove(src)
 	if(!bars.len)
 		LAZYREMOVE(user.progressbars, bar.loc)
-
-	if(client)
-		client.images -= bar
-	qdel(bar)
+	animate(bar, alpha = 0, time = 5)
+	qdel(bar, 5)
```

Comment:
It looks like this is supposed to delay the `qdel` by 5 deciseconds so the animation can finish, but that's not a feature of our `qdel` proc. Replacing this with the `QDEL_IN` macro would have the intended effect... 

... but that would still be wrong, because the code to remove the bar from the client's `images` has been removed, so it will linger on the client's screen (with 0 alpha, so, invisible) until the GC hard deletes it. Not good. You'd have to re-add the removal with a delay, either with a spawn (easy) or a timer (less straightforward, but probably ever so slightly more efficient).